sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.JsonSchema/src/SchemaRegistryJsonSchemaGenerator.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Data.SchemaRegistry;
+using System;
+
+namespace Microsoft.Azure.Data.SchemaRegistry.JsonSchema
+{
+    /// <summary>
+    /// This abstract class must be implemented in order to use the <see cref="SchemaRegistryJsonSerializer"/>. It allows
+    /// any available library to be used to generate schemas from .NET types and validate objects against schemas.
+    /// </summary>
+    /// <remarks>
+    /// Defining <see cref="GenerateSchemaFromObject(Type)"/> is required. However, defining <see cref="ValidateAgainstSchema(object, Type, string)"/>
+    /// is optional.
+    /// </remarks>
+    public abstract class SchemaRegistryJsonSchemaGenerator
+    {
+        /// <summary>
+        /// Validates that <paramref name="data"/> is valid according to <paramref name="schemaDefinition"/>.
+        /// </summary>
+        /// <param name="data">The data to use for serialization or the data that was deserialized.</param>
+        /// <param name="dataType">The type of the data to serialize or the type of the deserialized data.</param>
+        /// <param name="schemaDefinition">The JSON schema definition retrieved using <see cref="SchemaRegistryClient"/></param>
+        /// <returns> <c>true</c> if <paramref name="data"/> is valid, <c>false</c> otherwise.</returns>
+        public virtual bool ValidateAgainstSchema(Object data, Type dataType, string schemaDefinition)
```

Comment:
Also - the validation process in most of the libraries is done during deserialization using a validating reader (either under the hood or directly) (which is unfortunate since we can't support that scenario very easily here unless they re-serialize -> deserialize again with validation).

The validating reader collects the errors into a list and the guidance is to check that list and see all the errors, so exceptions would need to be thrown manually. Unless we want to consider having this method return a list of errors?

References:
- https://www.newtonsoft.com/jsonschema/help/html/ValidatingJson.htm
- https://github.com/RicoSuter/NJsonSchema/wiki/JsonSchemaValidator
- https://docs.json-everything.net/schema/basics/#schema-evaluation-2 OR https://docs.json-everything.net/schema/serialization/

Reply:
I think that makes sense, and I buy your rationale about an exception being better able to return rich information to the caller that a `bool` cannot.   

I do wonder, though, if that will be clear to developers implementing.   Sounds like a good "Hey Kryzsztof or Ted" email.

Reply:
> Unless we want to consider having this method return a list of errors?

`AggregateException`'s job is to represent one or more errors, wrapped up in a single exception.   So, I think we're good there.

Reply:
I wonder - if we do make this a "throw if invalid", should we reconsider making it `abstract` and force someone to choose to no-op things?